<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <param name="robot_description"
    textfile="$(find sensor_driver)/urdf/multivelodyne_3d.urdf" />
    
  <node name="robot_state_publisher" pkg="robot_state_publisher"
    type="robot_state_publisher" />
    
  <arg name="rviz" default="false" />


  <arg name="configfile" default="$(find sensor_driver)/config/configmultilidar-cwproject.xml" />
  <arg name="logdir" default="--log_dir=/home/zhubc/lab/code/20181108/catkin_ws/log" />
  <arg name="glogparam" default="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=5   $(arg logdir)" />

  <node pkg="sensor_driver" type="masternode" name="masternode" args="$(arg glogparam)" output="screen">
  	<param name="config_file" value="$(arg configfile)"/>
  </node>



  <node pkg="obstacle_detection" type="obstacle_detection_node" name="obstacle_detection_node" args="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=2   $(arg logdir)" output="screen">
  <param name="heightdiff_threshold" value="0.3"/>
  <param name="heightdiff_threshold_higher" value="0.5"/>
  <param name="N_SCANS" value="32"/>
  <param name="display_on" value="true"/>
  </node>


 <node pkg="sensor_driver" type="getmultirslidardata" name="getmultirslidardata" args="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=2   $(arg logdir)" output="screen">
  <param name="fix_frame" value="true"/>
  </node>
 

  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find obstacle_detection)/rviz_cfg/covgrid_mapping.rviz" />
  </group>

</launch>
